### PR TITLE
Remove all caching related functionality

### PR DESCRIPTION
**Pull request by dmaicher**
Fully remove all deprecated caching functionality

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -91,26 +74,15 @@ public function getConfigTreeBuilder(): TreeBuilder
...
-                            return $v;
-                        })
-                    ->end()
-                    ->children()
-                        // NEXT_MAJOR: remove option
-                        ->scalarNode('handler')->defaultValue('sonata.block.cache.handler.default')->end()
-                        // NEXT_MAJOR: remove option
-                        ->booleanNode('listener')->defaultTrue()->end()
-                    ->end()
+                // TODO: remove on 6.x
```
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php]: NEXT_MAJOR

Comment (dmaicher) [reply src/DependencyInjection/Configuration.php]: I guess then someone might remove it on 5.x already and think its some 4.x leftover? :smile:

**Comment by dmaicher** [comment]
The only remaining thing I'm unsure about is

https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Block/Service/MenuBlockService.php#L107

https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Block/Service/MenuBlockService.php#L72

Is this still useful? :thinking:

**Comment by dmaicher** [comment]
Ah and this piece of code here that deals with response TTLs:

https://github.com/sonata-project/SonataBlockBundle/blob/0b54efe3d3670e0644967753259db6ebf9c46ca1/src/Block/BlockRenderer.php#L104

useful? :thinking:

**Comment by VincentLanglet** [comment]
> The only remaining thing I'm unsure about is
> 
> https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Block/Service/MenuBlockService.php#L107
> 
> https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Block/Service/MenuBlockService.php#L72
> 
> Is this still useful? 🤔

According to the phpdoc it's cache-related https://github.com/sonata-project/SonataBlockBundle/blob/3d589d6bd26f9cab6bd26c2bbea60c04dcaad18a/src/Block/Service/AbstractBlockService.php#L51-L62
So it doesn't seems usefull anymore

**Comment by dmaicher** [comment]
@VincentLanglet yeah I believe all caching related things are actually useless now. Let me see if we need more deprecations on 4.x and remove them here.

File: src/Block/Service/AbstractBlockService.php [hunk 2]
```diff
@@ -48,19 +48,6 @@ public function renderResponse(string $view, array $parameters = [], ?Response $
         return $response;
     }
 
-    /**
-     * Returns a Response object that cannot be cacheable, this must be used if the Response is related to the user.
-     * A good solution to make the page cacheable is to configure the block to be cached with javascript ...
-     *
-     * @param array<string, mixed> $parameters
-     */
-    public function renderPrivateResponse(string $view, array $parameters = [], ?Response $response = null): Response
```
Comment (dmaicher) [inline src/Block/Service/AbstractBlockService.php:61]: todo: deprecate on 4.x

File: src/Block/Service/MenuBlockService.php [hunk 3]
```diff
@@ -104,7 +100,6 @@ public function configureSettings(OptionsResolver $resolver): void
     {
         $resolver->setDefaults([
             'title' => '',
-            'cache_policy' => 'public',
```
Comment (dmaicher) [inline src/Block/Service/MenuBlockService.php:109]: todo: deprecate on 4.x

File: src/Block/Service/AbstractBlockService.php [hunk 4]
```diff
@@ -48,31 +48,6 @@ public function renderResponse(string $view, array $parameters = [], ?Response $
...
-     * Returns a Response object that cannot be cacheable, this must be used if the Response is related to the user.
-     * A good solution to make the page cacheable is to configure the block to be cached with javascript ...
-     *
-     * @deprecated since sonata-project/block-bundle 4.x and will be removed in 5.0.
-     *
-     * NEXT_MAJOR: remove
-     *
-     * @param array<string, mixed> $parameters
-     */
-    public function renderPrivateResponse(string $view, array $parameters = [], ?Response $response = null): Response
```
Comment (dmaicher) [inline src/Block/Service/AbstractBlockService.php:61]: this is currently called on SonataAdmin. Will also look into making that compatible with 5.x.

Comment (VincentLanglet) [reply src/Block/Service/AbstractBlockService.php:61]: Yeah would be great to have sonataAdmin 4 compatible with both SonataBlock 4 and 5.

**Review by VincentLanglet — APPROVED** [review]

File: src/Block/Service/AbstractBlockService.php (hunk 4 above)
Comment (dmaicher) [reply src/Block/Service/AbstractBlockService.php:61]: See https://github.com/sonata-project/SonataAdminBundle/pull/7792

will also look into allowing 5.0 via composer on SonataAdmin

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thanks @dmaicher

**Comment by VincentLanglet** [comment]
@dmaicher Do you think there is still some PR to do on 4.x or we can create a release with all the deprecations ?

**Comment by dmaicher** [comment]
@VincentLanglet for now I think we are good on 4.x :blush: So creating a new release seems good to me.

Then we should also create a new release on SonataAdminBundle to have https://github.com/sonata-project/SonataAdminBundle/pull/7792

**Pull request merged**